### PR TITLE
PR template 

### PR DESCRIPTION
**Pull request by Ch0live**
## Description
Adds a PR template and the beginnings of a contributing doc. Contributing doc is a space for guidance on how to contribute. For now this is mostly focused on the rules we've enforced on the repository

## Changelog
- [x] Adds template for PRs in `.github/pull_request_template.md`
- [x] Adds `CONTRIBUTING.md` with info on enforced repo rules
- [x] Links `README.md`to `CONTRIBUTING.md`
- [x] Adds `requirements.txt` to backend space

### CODE REVIEW

File: backend/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+fastapi
+uvicorn
+mistralai
```
Comment (harryBedfordSL) [inline backend/requirements.txt:3]: should we specify versions here? Feel like I've seen that somewhere to make sure people in the future get a working version? wdyt?

Comment (DannyNicholas) [reply backend/requirements.txt:3]: From what I've read, it's good practice to specify a version. Can avoid dependency hell in future when some libraries have updated but others haven't.

It's also our way of saying: "this code has been proven to work successfully against these versions".

Comment (Ch0live) [reply backend/requirements.txt]: I agree - I've added versions (& updated the readme as it had a mistake in it)

**Review by bheyman-scottlogic — APPROVED** [review]
Looks good!

**Review by harryBedfordSL — APPROVED** [review]
lgtm

**Pull request merged**